waku/handshake.go

```diff
@@ -48,24 +95,34 @@ var keyFieldIdx = func() map[string]int {
 	return result
 }()
 
-func (o statusOptions) PoWRequirementF() float64 {
-	return math.Float64frombits(o.PoWRequirement)
+func (o statusOptions) PoWRequirementF() *float64 {
+	if o.PoWRequirement == nil {
+		return nil
+	}
+	result := math.Float64frombits(*o.PoWRequirement)
+	return &result
 }
 
 func (o *statusOptions) SetPoWRequirementFromF(val float64) {
-	o.PoWRequirement = math.Float64bits(val)
+	requirement := math.Float64bits(val)
+	o.PoWRequirement = &requirement
 }
 
 func (o statusOptions) EncodeRLP(w io.Writer) error {
 	v := reflect.ValueOf(o)
-	optionsList := make([]interface{}, 0, v.NumField())
+	var optionsList []interface{}
 	for i := 0; i < v.NumField(); i++ {
-		value := v.Field(i).Interface()
-		key, ok := idxFieldKey[i]
-		if !ok {
-			continue
+		field := v.Field(i)
+		if !field.IsNil() {
```

Comment:
```suggestion
		if field.CanInterface() {
```
Might be even safer as it will never panic.

Reply:
I think it looks like it does something slightly different, did you mean something like:
`if field.CanInterface() && !field.IsNil()` ? i.e is the field public and is not nil?

Reply:
Sorry, I misunderstood `CanInterface`. `!field.IsNil()` is good 👍